testbed/tests/window/test_window.py

```diff
@@ -138,13 +139,194 @@ async def test_move_and_resize(main_window, main_window_probe, capsys):
         finally:
             main_window.content = orig_content
 
-    async def test_full_screen(main_window, main_window_probe):
-        """Window can be made full screen"""
-        main_window.full_screen = True
-        await main_window_probe.wait_for_window("Full screen is a no-op")
+    @pytest.mark.parametrize(
+        "initial_state, final_state",
+        [
+            # Direct switch from NORMAL:
+            (WindowState.NORMAL, WindowState.FULLSCREEN),
+            (WindowState.NORMAL, WindowState.PRESENTATION),
+            # Direct switch from FULLSCREEN:
+            (WindowState.FULLSCREEN, WindowState.NORMAL),
+            (WindowState.FULLSCREEN, WindowState.PRESENTATION),
+            # Direct switch from PRESENTATION:
+            (WindowState.PRESENTATION, WindowState.NORMAL),
+            (WindowState.PRESENTATION, WindowState.FULLSCREEN),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "intermediate_states",
+        [
+            (
```

Comment:
It's not clear to me what this specific set of "intermediate" states are achieving. There's 2 non-trivial lists of intermediate states... why these in particular?

Reply:
I agree with you that these sets of intermediate states weren't testing anything useful, especially given the blocking nature of the Android API. So, I have removed one of the two cases, leaving the other to test that rapidly assigning new window state doesn't cause any unexpected problems on the Android backend.

Reply:
Ok... but again... I'm not clear on *what* this is testing. It's now a parameterised test with 1 parameterised value... and it's not clear at all what case it's testing, or why that specific case needs to be tested. 

Each of the individual transitions should be tested as part of the previous test; if there's a need to test a specific transition through an intermediate state, then that sequence should be tested (and it should be documented why it's an issue). 

Ideally, we'd test *all* possible intermediate states; however, I'll grant that 125 test cases is excessive here, so in the name of practicality, it's acceptable to cut to the subset that we believe to be an issue... but it should be clear which "set of three" is the problem.

(And, to be clear - the same issue applies to the desktop intermediate states test)

Reply:
Basically, I wanted to test that rapidly passing new states to `Window.state` setter doesn't cause any glitches, i.e., doing:
```python
def do_state_switching(self, widget, **kwargs):
    self.main_window.state = WindowState.PRESENTATION
    self.main_window.state = WindowState.FULLSCREEN
    self.main_window.state = WindowState.NORMAL
```
doesn't cause any unexpected glitch. The set of states, I have chosen to be intermediate_states for the test, is purely random and do not test for any specific "intermediate state transitional" issue.

However, on the desktop intermediate states test, the set of states chosen to be intermediate_states, are specifically chosen in specific order, and having specific number of states in each set. This is to trigger certain cases, which were causing glitches leading to test failure(on macOS & Gtk). So, keeping these 2 specific intermediate_states test cases is important to ensure any new future changes doesn't change any behavior that can lead to glitches.